lib/utils/matrix_sdk_extensions/flutter_matrix_dart_sdk_database/sqlite_ffi/io.dart

```diff
@@ -0,0 +1,46 @@
+import 'dart:ffi';
+import 'dart:io';
+import 'dart:math';
+
+import 'package:sqlite3/open.dart';
+
+void ffiInit() {
+  open.overrideFor(OperatingSystem.linux, sqlcipherOpen);
+}
+
+DynamicLibrary sqlcipherOpen() {
+  // Taken from https://github.com/simolus3/sqlite3.dart/blob/e66702c5bec7faec2bf71d374c008d5273ef2b3b/sqlite3/lib/src/load_library.dart#L24
```

Comment:
This needs much much more information why this is needed. The link only shows me that the exact same code is already in a different library? Why do we need it here then and can't just use what is there? This must be explained here in more detail

Reply:
Added the corresponding doc link in he file header.